### PR TITLE
#29420690: Modify Bootstrap Layouts Module

### PR DESCRIPTION
**Pull request by willjackson**
## Description
Add default paragraph template for when bootstrap_layout_classes widget is used on a paragraph entity.

## Teamwork Ticket(s)
- [#29420690: Modify Bootstrap Layouts Module](https://kanopi.teamwork.com/app/tasks/29420690)

## Testing
1. Pull down changes from this pull request.
2. Add/Edit page content with Text paragraph type.
3. Add custom styling via the Styles tab for the paragraph.
4. Save and view the page and verify DOM elements appear as expected.

## Deploy Notes
- `drush cr;`

### CODE REVIEW

File: bootstrap_layout_classes.module [hunk 1]
```diff
@@ -62,11 +63,52 @@ function bootstrap_layout_classes_entity_view_alter(array &$build, EntityInterfa
...
     }
   }
 }
 
+/**
+ * Implements hook_theme().
+ */
+function bootstrap_layout_classes_theme($existing, $type, $theme, $path) {
+  return [
+    'bootstrap_layout_classes' => [
```
Comment (thejimbirch) [inline bootstrap_layout_classes.module]: This should be `'paragraph__default' => [` for better developer visibility as we are overriding the paragraph template below.

File: bootstrap_layout_classes.module [hunk 2]
```diff
@@ -62,11 +63,52 @@ function bootstrap_layout_classes_entity_view_alter(array &$build, EntityInterfa
...
+      'variables' => [
+        'paragraph' => NULL,
+        'content' => [],
+        'attributes' => NULL,
+        'view_mode' => '',
+        'logged_in' => FALSE,
+        'is_admin' => FALSE,
+        'bootstrap_styles' => NULL,
+      ],
+      'template' => 'bootstrap_layout_classes',
```
Comment (thejimbirch) [inline bootstrap_layout_classes.module]: Then here you could just do `'base hook' => 'paragraph',`

File: templates/bootstrap_layout_classes.html.twig [hunk 3]
```diff
@@ -0,0 +1,98 @@
+{#
```
Comment (thejimbirch) [inline templates/bootstrap_layout_classes.html.twig]: Then this file would be named `paragraph--default.html.twig`

**Review by thejimbirch — COMMENTED** [review]
General questions:

Does this add any bloat/performance issues if I were to use the field on a node, and not have paragraphs installed?

Which leads me to, should it be in a submodule?

Forward thinking (not in scope).  Could this method be used for blocks/layout builder?  We'd probably only want `column_classes` there.

File: bootstrap_layout_classes.module (hunk 1 above)
Comment (willjackson) [reply bootstrap_layout_classes.module]: @thejimbirch - Great thought; the theme hook has been updated!

File: templates/bootstrap_layout_classes.html.twig (hunk 3 above)
Comment (willjackson) [reply templates/bootstrap_layout_classes.html.twig]: @thejimbirch - The template name has been updated as well.

File: bootstrap_layout_classes.module (hunk 2 above)
Comment (willjackson) [reply bootstrap_layout_classes.module]: @thejimbirch - An error was encountered with `'base hook' => 'paragraph',` in the hook_theme(), related to `paragraphs_theme_suggestions_paragraph()` with the `$variables['elements']` not having the intended values.  It's likely something that's being overlooked, but for the time being, reverted to not using the `base hook` property.

**Comment by willjackson** [comment]
> Does this add any bloat/performance issues if I were to use the field on a node, and not have paragraphs installed?
> 
> Which leads me to, should it be in a submodule?

@thejimbirch - There should be no bloat/performance issues with using across another entity. However, it must be extended to include additional templates for each entity. Depending on the intended usage, it could be moved into a submodule or refactor the main module to make it more flexible.  Hope this helps, but let me know if there are any questions!

File: bootstrap_layout_classes.module (hunk 2 above)
Comment (thejimbirch) [reply bootstrap_layout_classes.module]: Makes sense, thanks.  The example I was looking at didn't have the variables.

**Comment by thejimbirch** [comment]
Thanks for the clarity!

**Review by kmurphychi247 — APPROVED** [review]

**Pull request merged**